src/comodo/jaxsimSimulator/jaxsimSimulator.py

```diff
@@ -0,0 +1,465 @@
+            self._recorder = MujocoVideoRecorder(
+                model=self._mj_model_helper.model,
+                data=self._mj_model_helper.data,
+                fps=30,
+            )
+
+        self._is_initialized = True
+
+    def set_input(self, input: npt.ArrayLike) -> None:
+        self._tau = jnp.array(input)
+
+    def step(self, n_step: int = 1) -> None:
+        assert (
+            self._is_initialized
+        ), "Simulator is not initialized, call load_model first."
+
+        for _ in range(n_step):
+            if self._contact_model_type is ContactModelEnum.VISCO_ELASTIC:
+                self._data, _ = jaxsim.rbda.contacts.visco_elastic.step(
+                    model=self._model,
+                    data=self._data,
+                    dt=self._dt,
+                    joint_force_references=self._tau,
+                    link_forces=None,
+                )
+
+            else:
+                # All other contact models
+                self._data, self._integrator_state = js.model.step(
+                    model=self._model,
+                    data=self._data,
+                    dt=self.dt,
+                    integrator=self._integrator,
+                    integrator_state=self._integrator_state,
+                    joint_forces=self._tau,
+                    link_forces=None,
+                )
+
+                self._link_contact_forces = js.model.link_contact_forces(
+                    model=self._model,
+                    data=self._data,
+                    joint_force_references=self._tau,
+                )
+
+            current_time_ns = self._data.time_ns.astype(int)
+
+            match self._visualization_mode:
+                case "record":
+                    if current_time_ns - self._last_rendered_t_ns >= int(
+                        1e9 / self._recorder.fps
+                    ):
+                        self._record_frame()
+                        self._last_rendered_t_ns = current_time_ns
+
+                case "interactive":
+                    if current_time_ns - self._last_rendered_t_ns >= int(
+                        1e9 / self._viz_fps
+                    ):
+                        self._render()
+                        self._last_rendered_t_ns = current_time_ns
+                case None:
+                    pass
+
+    def get_state(self) -> Tuple[npt.ArrayLike, npt.ArrayLike, npt.ArrayLike]:
+        assert (
+            self._is_initialized
+        ), "Simulator is not initialized, call load_model first."
+
+        s = np.array(self._data.joint_positions())
+        s_dot = np.array(self._data.joint_velocities())
+        tau = np.array(self._tau)
+
+        return s, s_dot, tau
+
+    def close(self) -> None:
+        pass
+
+    # ==== Properties and public methods ====
+
+    @property
+    def feet_wrench(self) -> npt.ArrayLike:
+        assert (
+            self._is_initialized
+        ), "Simulator is not initialized, call load_model first."
```

Comment:
Mmh this is the right use-case of the lazy initialization that I explained to you yesterday. Furthermore, in this case, you don't only need that the simulation was initialized. It also needs to be stepped once. Probably you can just check if `link_contact_forces` is None. If it is not, it means that at least one step occurred.

Reply:
Yes you're right I can optimize this!

Reply:
Done in 92f3d5e703956a1e4c3330393bccd1e58497cf7d